compiler/rustc_borrowck/src/def_use.rs

```diff
@@ -52,12 +52,16 @@ pub fn categorize(context: PlaceContext) -> Option<DefUse> {
         PlaceContext::NonMutatingUse(NonMutatingUseContext::ShallowBorrow) |
         PlaceContext::NonMutatingUse(NonMutatingUseContext::UniqueBorrow) |
 
+        // `PlaceMention` and `AscribeUserType` both evaluate the place, which must not
+        // contain dangling references.
+        PlaceContext::NonUse(NonUseContext::PlaceMention) |
+        PlaceContext::NonUse(NonUseContext::AscribeUserTy) |
```

Comment:
I don't know borrowck, so someone else will have to review this part.
But doesn't this do a check that the place is actually initialized? Just to avoid UB it would be sufficient to check that the place is live, but maybe borrowck does not have such a check.

Reply:
Checking that the place is initialized will break a lot of code, because we allow binding moved-from places to `_`. For instance in test `src/test/ui/binding/issue-53114-borrow-checks.rs`.

Reply:
Yes an init check is not needed. I just thought `DefUse::Use` does an init check. But it seems I am wrong about that?